lib/shortcuts/stream-lambda.js

```diff
@@ -59,6 +78,25 @@ class StreamLambda extends Lambda {
         StartingPosition
       }
     };
+    if (FilterCriteria) {
+      if (Object.prototype.toString.call(FilterCriteria) !== '[object Object]'){
+        throw new Error('`FilterCriteria` must be a JSON-like object');
+      }
+      if (!(FilterCriteria.Filters && Array.isArray(FilterCriteria.Filters))){
+        throw new Error('`FilterCriteria` must contain property `Filter` of type array');
+      }
+      for (const filter of FilterCriteria.Filters){
```

Comment:
This is a fail fast strategy, do you think it might be better to collect all failures and then fail after?

Reply:
I think fail-fast is fine here, usage of this field is well documented by AWS so we shouldn't expect to hit these errors often anyway

Reply:
Not sure I agree, but this is fine.